src/exojax/spec/moldb.py

```diff
@@ -623,3 +623,496 @@ def search_molecid(molec):
+        ionE2 = jnp.array(list(map(lambda x: self.ipccd[self.ipccd['ielem']==x].iat[0, 6], self.ielem)))
+        self.ionE = ionE * np.where(self.iion==1, 1, 0) + ionE2 * np.where(self.iion==2, 1, 0)
+        self.solarA = jnp.array(list(map(lambda x: self.ipccd[self.ipccd['ielem']==x].iat[0, 4], self.ielem)))
+        self.atomicmass = jnp.array(list(map(lambda x: self.ipccd[self.ipccd['ielem']==x].iat[0, 5], self.ielem)))
+        
+            
+            
+    #End of the CONSTRUCTOR definition ↑
+
+
+
+
+    #Defining METHODS ↓
+    
+    
+    
+    def masking(self,mask):
+        """applying mask and (re)generate jnp.arrays
+        
+        Args:
+           mask: mask to be applied. self.mask is updated.
+
+        Note:
+           We have nd arrays and jnp arrays. We apply the mask to nd arrays and generate jnp array from the corresponding nd array. For instance, self._A is nd array and self.A is jnp array.
+
+        """
+        #numpy float 64 Do not convert them jnp array
+        self.nu_lines = self.nu_lines[mask]
+        self.Sij0 = self.Sij0[mask]
+        self._A=self._A[mask]
+        self._elower=self._elower[mask]
+        self._eupper=self._eupper[mask]
+        self._gupper=self._gupper[mask]
+        self._jlower=self._jlower[mask]
+        self._jupper=self._jupper[mask]
+        self._QTmask=self._QTmask[mask]
+        self._ielem=self._ielem[mask]
+        self._iion=self._iion[mask]
+        self._gamRad=self._gamRad[mask]
+        self._gamSta=self._gamSta[mask]
+        self._vdWdamp=self._vdWdamp[mask]
+
+        #jnp arrays
+        self.dev_nu_lines=jnp.array(self.nu_lines)
+        self.logsij0=jnp.array(np.log(self.Sij0))
+        self.A=jnp.array(self._A)
+        self.elower=jnp.array(self._elower)
+        self.eupper=jnp.array(self._eupper)
+        self.gupper=jnp.array(self._gupper)
+        self.jlower=jnp.array(self._jlower,dtype=int)
+        self.jupper=jnp.array(self._jupper,dtype=int)
+        
+        self.QTmask=jnp.array(self._QTmask,dtype=int)
+        self.ielem=jnp.array(self._ielem,dtype=int)
+        self.iion=jnp.array(self._iion,dtype=int)
+        self.gamRad=jnp.array(self._gamRad)
+        self.gamSta=jnp.array(self._gamSta)
+        self.vdWdamp=jnp.array(self._vdWdamp)
+
+
+
+    def Atomic_gQT(self, atomspecies):
+        """Select grid of partition function especially for the species of interest
+        
+        Args:
+            atomspecies: species e.g., "Fe 1", "Sr 2", etc.
+            
+        Returns:
+            gQT: grid Q(T) for the species
+        
+        """
+        #if len(atomspecies.split(' '))==2:
+        atomspecies_Roman = atomspecies.split(' ')[0] + '_' + 'I'*int(atomspecies.split(' ')[-1])
+        gQT = self.gQT_284species[ np.where(self.pfdat['T[K]']==atomspecies_Roman) ][0]
+        return gQT
+    
+    
+    
+    def QT_interp(self, atomspecies, T, Irwin=False):
```

Comment:
We will use this funtion in an HMC. ”if” statement is incompatible with JIT/XLA. Consider separating them into multiple functions or other solutions.